t/op/array.t

```diff
@@ -707,3 +707,13 @@ fresh_perl_is('my @x;$x[0] = 1;shift @x;$x[22] = 1;$x[25] = 1;','',
   {}, 'unshifting and growing an array initializes trailing elements');
 
 "We're included by lib/Tie/Array/std.t so we need to return something true";
+
+# GH #23447 - ensure that future optimizations don't break behaviour
+{
+    my @x = "a" .. "d"; sub f {} my $y = @x[2, 3, f()];
+    is $y, d, 'Non-returning final sub element in array slice in scalar context';
```

Comment:
It might not have "return" but it does return an empty list.

Maybe "Empty list returning final sub element..."

Reply:
How about the changed text: "Trailing empty list return in..."?